addons/mail/models/discuss/mail_guest.py

```diff
@@ -135,7 +135,7 @@ def _init_messaging(self):
         # sudo: mail.guest - guest reading their own id/name/channels
         guest_sudo = self.sudo()
         return {
-            'Thread': guest_sudo.channel_ids.sudo(False)._channel_info(),
+            'Thread': guest_sudo.channel_ids._filter_for_init_messaging()._channel_info(),
```

Comment:
rather than post filtering, it would be better to adapt the query/domain accordingly and only fetch what is necessary from the database